drivers/serial/uart_stm32.c

```diff
@@ -1656,7 +1656,9 @@ static int uart_stm32_init(const struct device *dev)
 	}
 
 	/* Set the default baudrate */
-	uart_stm32_set_baudrate(dev, data->baud_rate);
+	if (data->baud_rate) {
+		uart_stm32_set_baudrate(dev, data->baud_rate);
+	}
```

Comment:
Could be helpful to have `LOG_WARN` or `__ASSERT_NO_MSG` for cases when `CONFIG_UART_USE_RUNTIME_CONFIGURE` is disabled and `data->baud_rate` is not provided.

Reply:
If needed, this can be resolved build time, I have a special macro for such special requirements : )